### PR TITLE
feat(popup-menu): allow grouping header entries

### LINKED ISSUE #896: Support grouping of popup menu header entries
### Is your feature request related to a problem? Please describe.

The popup menu allows for the (visual) grouping of entries but not of header entries.

### Describe the solution you'd like

Popup menu header entries should support a `group` property to group them visually (including dividers).

### Describe alternatives you've considered

Not supporting groups.

### Additional context

Required in the context of https://github.com/camunda/product-hub/issues/2006

### PR DESCRIPTION
**Pull request by philippfromme**
Try out via

```
npx @bpmn-io/sr bpmn-io/bpmn-js#popup-menu-header-entry-groups-example -l bpmn-io/diagram-js#popup-menu-header-entry-groups
```

brave_h88fABOpo3

Closes #896

### CODE REVIEW

**Comment by nikku** [comment]
How can I try this E2E? :slightly_smiling_face:

**Comment by philippfromme** [comment]
> How can I try this E2E? 🙂

`npx @bpmn-io/sr bpmn-io/bpmn-js#popup-menu-header-entry-groups-example -l bpmn-io/diagram-js#popup-menu-header-entry-groups`

File: lib/features/popup-menu/PopupMenuHeader.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    title
+  } = props;
+
+  const groups = useMemo(() => groupEntries(headerEntries), [ headerEntries ]);
+
+  return html`
+    <div class="djs-popup-header">
+      <h3 class="djs-popup-title" title=${ title }>${ title }</h3>
+      ${ groups.map((group, index) => html`
+        ${ index > 0 && html`<div class="divider" />` }
```
Comment (nikku) [inline lib/features/popup-menu/PopupMenuHeader.js:37]: Works on my machine; only thing I see is that adding the divider is different across vertical entries (organized in `div` groups) and horizontal header entries (organized in flat list).

Did you consider stacking in `div` for horizontal grouping?

Comment (philippfromme) [reply lib/features/popup-menu/PopupMenuHeader.js:37]: No I didn't. Let me me look into that.

File: assets/diagram-js.css [hunk 2]
```diff
@@ -568,6 +569,29 @@ marker.djs-dragger tspan {
...
+  padding: 0;
+}
+
+.djs-popup-header-group .entry {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+}
+
+.djs-popup-header-group:not(:last-child):after {
```
Comment (nikku) [inline assets/diagram-js.css:587]: You could use the [next sibling selector](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors/Combinators#next-sibling_combinator):

```suggestion
.djs-popup-header-group + .djs-popup-header-group:before {
```

Comment (philippfromme) [reply assets/diagram-js.css:587]: You don't like my 3 pseudo selector construct? 😢

Comment (philippfromme) [reply assets/diagram-js.css]: Adjusted ✅

**Review by nikku — APPROVED** [review]
Great improvement. Cool to see header and body entries are now en par.

**Pull request merged**